File: kornell-api/src/main/scala/kornell/server/util/ValueFactory.scala
package kornell.server.util

import com.google.web.bindery.autobean.vm.AutoBeanFactorySource
import kornell.core.value.{ValueFactory => KVF}

object ValueFactory {
   val valueFactory = AutoBeanFactorySource.create(classOf[KVF])
   
   def newDate = valueFactory.newDate().as()

}

File: kornell-core/src/main/java/kornell/core/value/ValueFactory.java
package kornell.core.value;

import com.google.web.bindery.autobean.shared.AutoBean;
import com.google.web.bindery.autobean.shared.AutoBeanFactory;
import com.google.web.bindery.autobean.shared.AutoBeanFactory.Category;


@Category(DateCategory.class)
public interface ValueFactory extends AutoBeanFactory {
	AutoBean<Date> newDate();
}
